### PR TITLE
Module dependency fix

### PR DESCRIPTION
**Pull request by cabutlermit**
#### Developer Checklist

- [X] The README contains the correct list of dependencies, inputs, and outputs (e.g., `terraform-docs markdown .` has been run and copied)
- [X] Any special variables have values configured in the Terraform Cloud workspace
- [N/A] Stakeholder approval has been confirmed (or is not needed)

#### What does this PR do?

The first version of this code had a dependency problem (it was attempting to create workspaces before the branches in the repository were ready for linking). This addresses that dependency.

Additionally, the work to fix that highlighted the fact that the first version of this code did not configure and version/source constrain the `tfe` and `github` Providers correctly. This fixes that too.

#### Helpful background context

This is mostly just clean up to address a few glitches in the first draft of this code.

#### What are the relevant tickets?

https://mitlibraries.atlassian.net/browse/IN-428

#### Requires Database Migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Comment by cabutlermit** [comment]
The Terraform Cloud automatic plan failed because I was moving around provider configuration (to be expected). I have not run a `terraform destroy` yet with the code in the main branch, but I will do that after a review of this code and before I merge this PR.

File: workspaces.tf [hunk 1]
```diff
@@ -10,25 +10,20 @@ module "gh_apps_vars" {
...
-  gh_token           = var.gh_token
-  tfc_token          = var.tfc_token
+  source             = "./modules/tfc"
+  aws_ou             = "workloads"
+  gh_repo            = module.gh_apps_vars.repo_id
+  gh_organization    = "MITLibraries"
+  app_name           = "apps-vars"
+  environments       = ["prod", "stage"]
+  tfc_organization   = "MITLibraries"
+  tfc_version        = "1.0.4"
```
Comment (zotoMIT) [inline workspaces.tf:23]: Does it make sense to not lock this in the workspace creation, so instead it picks up the most recent version? Instead, we could rely on our versions.tf terraform version to bring us down to what we need i think?  I dont actually know what would be best here.

**Review by zotoMIT — DISMISSED** [review]
lgtm

File: workspaces.tf (hunk 1 above)
Comment (cabutlermit) [reply workspaces.tf:23]: I think it makes more sense to NOT set this the way I did here. I was waffling on this and in the end left it here. I'll clean this up shortly.

**Comment by cabutlermit** [comment]
@zotoMIT I have address the terraform version constraint (and also dealt with some problems related to `terraform destroy`. You'll need to re-review this. Thanks!

**Review by zotoMIT — APPROVED** [review]
Looks good. I think removing the version from here so it autoselects the most recent is good.

**Pull request merged**